addons/crm/models/crm_lead.py

```diff
@@ -1165,18 +1165,58 @@ def _get_rainbowman_message(self):
+                    SUM(CASE WHEN date_closed::date = CURRENT_DATE AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_today,
+                    SUM(CASE WHEN country_id = %(country_id)s AND team_id = %(team_id)s THEN 1 ELSE 0 END) as count_country_closed_year,
+                    SUM(CASE WHEN partner_id = %(source_id)s AND team_id = %(team_id)s THEN 1 ELSE 0 END) as count_source_closed_year,
+                    MIN(CASE WHEN date_closed >= CURRENT_DATE - INTERVAL '30 days' AND team_id = %(team_id)s THEN day_close ELSE 1000000 END) as min_day_close_30,
+
+                    SUM(CASE WHEN date_closed::date = (CURRENT_DATE - INTERVAL '1 day')  AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_yesterday,
+                    SUM(CASE WHEN date_closed::date = (CURRENT_DATE - INTERVAL '2 days') AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_minus2day,
+                    SUM(CASE WHEN date_closed::date = (CURRENT_DATE - INTERVAL '3 days') AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_minus3day
+                FROM crm_lead
+                WHERE
+                    type='opportunity'
+                AND
+                    active = True
+                AND
+                    probability = 100
+                AND
+                    DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', CURRENT_DATE)
+                AND
+                    (user_id = %(user_id)s OR team_id = %(team_id)s)
+            """
+            self.env.cr.execute(query, {
+                'user_id': self.user_id.id,
+                'team_id': self.team_id.id,
+                'country_id': self.country_id.id or -1,
+                'source_id': self.source_id.id or -1,
+            })
+            query_result = self.env.cr.dictfetchone()
+
+            if query_result['count_user_closed_year'] == 1:
                 return _('Go, go, go! Congrats for your first deal.')
+            elif query_result['count_user_closed_today'] == 5:
+                return _('You\'re on fire! Fifth deal won today 🔥')
+            elif query_result['count_country_closed_year'] == 1 and self.country_id:
+                return _('You just expanded the map! First win in %s.', self.country_id.name)
+            elif query_result['count_source_closed_year'] == 1 and self.source_id:
+                return _('Boom! First win from %s - keep it coming!', self.source_id.name)
+            elif len(self.message_ids) >= 25:
+                return _('Phew, that took some effort — but you nailed it. Good job!')
+            elif query_result['count_user_closed_yesterday'] and query_result['count_user_closed_minus2day'] and not query_result['count_user_closed_minus3day']:
+                return _('You\'re on a winning streak. 3 deals in 3 days, congrats!')
+            elif query_result['min_day_close_30'] == self.day_close:
+                return _('Wow, that was fast. That deal didn’t stand a chance!')
```

Comment:
A bit annoying during testing, as you get this message every single time you close a deal.
(Unless I wait for a day).

Don't we want to show that message if you *beat* the best time, instead of matching it? Otherwise it could get a bit annoying.